utils/vim/syntax/swift.vim

```diff
@@ -120,10 +120,8 @@ syn region swiftType contained contains=swiftType,swiftParamDelim
 syn match swiftParamDelim contained
       \ /,/
 " <Generic Clause> (generics)
-syn region swiftTypeParameters contained contains=swiftArchetype,swiftConstraint
+syn region swiftTypeParameters contained contains=swiftVarName,swiftConstraint
       \ start="<" end=">"
-syn match swiftArchetype contained skipwhite nextgroup=swiftTypeDeclaration
```

Comment:
Is this code trying to match the conformance constraints?

```
func foo<T : X, T.U : Y>() {}
         ^^^^^  ^^^^^^^^
```

The constrained type can be a nested one, so I'm not sure that `swiftVarName` would work.